client/pages/newTask.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import parseRoute from '../lib/parse-route';
+
+export default class NewTask extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      taskName: '',
+      statusId: '',
+      dateCreated: '',
+      deadline: '',
+      userId: '',
+      description: '',
+      projectId: ''
+    };
+    this.handleChangeTaskName = this.handleChangeTaskName.bind(this);
+    this.handleChangeDateCreated = this.handleChangeDateCreated.bind(this);
+    this.handleChangeDeadline = this.handleChangeDeadline.bind(this);
+    this.handleChangeDescription = this.handleChangeDescription.bind(this);
+    this.handleChangeStatus = this.handleChangeStatus.bind(this);
+    this.handleChangeUserId = this.handleChangeUserId.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const route = parseRoute(window.location.hash);
+    this.setState({ projectId: Number(route.params.get('projectId')) });
+  }
```

Comment:
Your App should just pass this info in a prop so the component doesn't need to "know about" what the hash is.